File: DSA/Aarush_590017879/Day_30/Q1_leetcode.cpp
#include <iostream>
#include <stack>
#include <algorithm>
using namespace std;

string removeDuplicates(string s) {
    stack<char> st;

    for (char ch : s) {
        if (!st.empty() && st.top() == ch) {
            st.pop();
        } else {
            st.push(ch);
        }
    }

    string result = "";

    while (!st.empty()) {
        result += st.top();
        st.pop();
    }

    reverse(result.begin(), result.end());

    return result;
}

int main() {
    string s;
    cin >> s;

    cout << removeDuplicates(s) << endl;

    return 0;
}

File: DSA/Navneetpant_590011569/Day_30/Q1_leetcode.java
class Solution {
    public String removeDuplicates(String s) {
        Stack <Character > stack = new Stack <> ();
        for (int i = 0 ; i < s.length() ; i++ ) {
            if (stack.isEmpty()){
                stack.push(s.charAt(i));
            }
            else if (stack.peek()!=s.charAt(i)){
                stack.push(s.charAt(i));
            }
            else if (stack.peek()==s.charAt(i)){
                stack.pop();
            }
        }
        
        StringBuilder sb = new StringBuilder();
        if (stack.size()==0) {
            return sb.toString();
        }
        while (!stack.isEmpty()) {
            sb.append(stack.pop());
        }
        sb.reverse();
        return sb.toString();
         
    }
}